### ISSUE TITLE
Query logs

### ISSUE THREAD

**Issue by marcelloinfoweb** [folbricht/routedns#409]
How is it possible to show a log of requests in the terminal in real time?

**Comment by folbricht** [comment]
I wanted to add this at some point anyway. Would you be able to try out the `issue-409` branch? See https://github.com/folbricht/routedns/pull/410 for an example.

**Comment by cbuijs** [comment]
You can already do so with the syslog stuff?

**Comment by folbricht** [comment]
True, it's been a while and I completely forgot about that, doh. The new one is a bit easier to use but this does look unnecessary. Should probably keep just syslog for logging and not merge #410

**Comment by cbuijs** [comment]
No worries. There is a case for both anyway ;-)

**Comment by marcelloinfoweb** [comment]
> I wanted to add this at some point anyway. Would you be able to try out the `issue-409` branch? See #410 for an example.

I depend on a binary, I can't compile it.

**Comment by marcelloinfoweb** [comment]
Error routedns[7180]: Error: unsupported group type 'query-log' for group 'query-log'

> I wanted to add this at some point anyway. Would you be able to try out the `issue-409` branch? See #410 for an example.

Error: routedns[7180]: Error: unsupported group type 'query-log' for group 'query-log'

**Comment by folbricht** [comment]
That won't work for you unless you can compile it from the branch. If you can't compile it, you could try out `syslog` instead

**Comment by marcelloinfoweb** [comment]
> That won't work for you unless you can compile it from the branch. If you can't compile it, you could try out `syslog` instead. There's an example in https://github.com/folbricht/routedns/blob/master/cmd/routedns/example-config/syslog.toml

I did it, I compiled the branch and now I have a log, but it doesn't show the blocked and/or released lists.

**Comment by folbricht** [comment]
`syslog` does log responses as well if that's what you're looking for.

**Comment by cbuijs** [comment]
If you want to log the blocks/allows. You need to start with `-l5` or `-l6` as parameter.

I start RouteDNS as follows and have all the logging in a file (`routedns.log`):
`./routedns -l6 routedns.toml >> routedns.log 2>&1 & disown`

Or get everything into syslog using logger:
`./routedns -l6 routedns.toml 2>&1 | /usr/bin/gawk '{print substr($0, index($0, $3))}' | /usr/bin/logger -t "ROUTEDNS" & disown`

You can play with the log levels to get more/less logging. I think from `-l5` on, blocking is logged.

Above examples make RouteDNS run in the background and it stays there running when you log off.

**Comment by folbricht** [comment]
There is a `query-log` group type that writes one line per query to STDOUT or to a file, with `output-file` and `output-format` set to `text` or `json`. It can be placed anywhere in a chain. See `example-config/query-log.toml`.

The `syslog` element remains available for sending queries and responses to a syslog server.

Closing as done.
